File: src/breslin/clientside/application/partido/states/application_partido_states.php
var ApplicationMainPartido = new Class(
{

Extends: ApplicationMain,
	
initialize: function()
{
},

enter: function(applicationPartido)
{
	this.parent(applicationPartido);
},

execute: function(applicationPartido)
{
	this.parent(applicationPartido);

	//join game B
	if (applicationPartido.mButtonHit == applicationPartido.mButtonJoinGameB)
	{
	        applicationPartido.mButtonHit = 0;
		applicationPartido.setGame(new GamePartido(applicationPartido));
 		applicationPartido.sendJoinGame('2');
                applicationPartido.mStateMachine.changeState(applicationPartido.mApplicationPlay);
	}
},

exit: function(applicationPartido)
{
	this.parent(applicationPartido);
}

});

var ApplicationPlayPartido = new Class(
{
Extends: ApplicationPlay,
	
initialize: function()
{
},

enter: function(applicationPartido)
{
	applicationPartido.mPlayingGame = true;
	applicationPartido.mSentLeaveGame = false;
},

execute: function(applicationPartido)
{
	//check for logout as well....
   	if (applicationPartido.mLoggedIn == false)
        {
                applicationPartido.mStateMachine.changeState(applicationPartido.mApplicationLogin);
	}
	
	if (applicationPartido.mKeyArray[27] && applicationPartido.mSentLeaveGame == false)
        {
		//check to see if in battle....
		if (applicationPartido.getGame().mStateMachine.currentState() == applicationPartido.getGame().mGamePlayPartidoBattle)
		{	
			applicationPartido.mAnswerTime = 3;
			applicationPartido.mStringAnswer = 'esc';
			applicationPartido.sendAnswer();	
		}
		else
		{
			applicationPartido.log('send_leave_game....');
       			message = '';
        		applicationPartido.mNetwork.mSocket.emit('send_leave_game', message);
			applicationPartido.mSentLeaveGame = true;
		}
		applicationPartido.mKeyArray[27] = false;
        }

	if (applicationPartido.mLeaveGame)
	{
		applicationPartido.log('leave game');	
		applicationPartido.mSentLeaveGame = false;
		if (applicationPartido.mLoggedIn)
		{
                	applicationPartido.mStateMachine.changeState(applicationPartido.mApplicationMain);
		}
		else
		{
                	applicationPartido.mStateMachine.changeState(applicationPartido.mApplicationLogin);
		}
	}
        else
        {
                //game
		if (applicationPartido.getGame())
		{
                	applicationPartido.getGame().processUpdate();
		}
        }
},

exit: function(applicationPartido)
{
        applicationPartido.mPlayingGame = false;
	applicationPartido.mLeaveGame = false;
	if (applicationPartido.getGame())
	{
		applicationPartido.getGame().remove();
		applicationPartido.setGame(0);
	}
}

});

File: src/breslin/clientside/application/partido/states/applicationPartidoStates.cpp
//header
#include "applicationPartidoStates.h"

//log
#include "../../../tdreamsock/dreamSockLog.h"

//game
#include "../../../application/partido/applicationPartido.h"

//game
#include "../../../game/partido/gamePartido.h"

//shape
#include "../../../shape/shape.h"

//utility
#include <math.h>

//command
#include "../../../command/command.h"

#include "../../../game/partido/states/gamePartidoStates.h"

/******************** GLOBAL_PARTIDO_APPLICATION *****************/

GLOBAL_PARTIDO_APPLICATION* GLOBAL_PARTIDO_APPLICATION::Instance()
{
  static GLOBAL_PARTIDO_APPLICATION instance;
  return &instance;
}
void GLOBAL_PARTIDO_APPLICATION::enter(ApplicationPartido* application)
{
}
void GLOBAL_PARTIDO_APPLICATION::execute(ApplicationPartido* application)
{
}
void GLOBAL_PARTIDO_APPLICATION::exit(ApplicationPartido* application)
{
}
bool GLOBAL_PARTIDO_APPLICATION::onLetter(ApplicationPartido* application, Letter* letter)
{
        return true;
}

/******************** INIT_PARTIDO_APPLICATION *****************/

INIT_PARTIDO_APPLICATION* INIT_PARTIDO_APPLICATION::Instance()
{
  static INIT_PARTIDO_APPLICATION instance;
  return &instance;
}
void INIT_PARTIDO_APPLICATION::enter(ApplicationPartido* application)
{
}
void INIT_PARTIDO_APPLICATION::execute(ApplicationPartido* applicationPartido)
{
	if (applicationPartido->mPlayingGame)
	{
                applicationPartido->mPartidoStateMachine->changeState(PLAY_PARTIDO_APPLICATION::Instance());
	}
}
void INIT_PARTIDO_APPLICATION::exit(ApplicationPartido* application)
{
}
bool INIT_PARTIDO_APPLICATION::onLetter(ApplicationPartido* application, Letter* letter)
{
        return true;
}

/******************** PLAY_PARTIDO_APPLICATION *****************/

PLAY_PARTIDO_APPLICATION* PLAY_PARTIDO_APPLICATION::Instance()
{
        static PLAY_PARTIDO_APPLICATION instance;
        return &instance;
}
void PLAY_PARTIDO_APPLICATION::enter(ApplicationPartido* application)
{
}
void PLAY_PARTIDO_APPLICATION::execute(ApplicationPartido* applicationPartido)
{
	//Should this be in game states...
/*
        if (applicationPartido->mKeyArray[27]) //esc
        {
                //check to see if in battle....
                if ( applicationPartido->getGame()->mPartidoStateMachine->currentState() == BATTLE_GAME::Instance() )
                {
                        applicationPartido->mAnswerTime = 2001;
                        applicationPartido->mStringAnswer = "esc";
                        LogString("sendAnswer via esc");
                        applicationPartido->sendAnswer();
                }
	}
*/
}
void PLAY_PARTIDO_APPLICATION::exit(ApplicationPartido* applicationPartido)
{
}
bool PLAY_PARTIDO_APPLICATION::onLetter(ApplicationPartido* application, Letter* letter)
{
        return true;
}


